src/audio_pixel.h

```diff
@@ -67,15 +106,54 @@ class audio_pixel_block_t {
 // and is able to interpolate between them to 
 // get audio pixels at any resolution inbetween
 class audio_pixel_mipmap_t {
-    audio_pixel_mipmap_t(MediaItem_Take* take);
-
-    void update(); // update all blocks
+    audio_pixel_mipmap_t(MediaItem_Take* take)
+        :m_accessor(safe_audio_accessor_t(take)),
+        m_take(take)
+    {
+        if (!m_accessor.is_valid()) { std::cerr << "Invalid audio accessor used for audio block." << std::endl; }
+    };
+
+    void update() { // update all blocks
+        // MUST BE CALLED BY MAIN THREAD
+        // due to reaper api calls: AudioAccessorUpdate
+        AudioAccessor* safe_accessor = m_accessor.get();
+
+        if (AudioAcessorStateChange(safe_accessor)) {
```

Comment:
you should abstract all of the logic under here to a member function, so we can fill up the blocks before the run loop if we need to